### PR TITLE
chore: generate rc commits spreadsheet

### PR DESCRIPTION
**Pull request by sethkfman**
## **Description**

This PR includes a script to generate the changes between two branches. It will create a `csv` that is use by QA during release testing. 

## **Related issues**

Fixes:

## **Manual testing steps**

1. Fetch the latest from the repo
2. Run command `./generate-rc-commits vx.x.x release/x.x.x`
3. Look for csv called commits.csv 

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: scripts/generate-rc-commits.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    'Owen Craston',
+    'Gustavo Antunes',
+    'Monte Lai',
+    'Daniel Rocha',
+    'Howard Braham',
+    'Kate Johnson',
+    'Brian Bergeron',
+  ],
+  'Extension UX': ['David Walsh', 'vthomas13', 'Nidhi Kumari', 'Victor Thomas'],
+  'Extension Platform': [
```
Comment (cortisiko) [inline scripts/generate-rc-commits.js:24]: any reason why we need the extension platform team?

File: scripts/generate-rc-commits.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  Accounts: [
+    'Owen Craston',
+    'Gustavo Antunes',
+    'Monte Lai',
+    'Daniel Rocha',
+    'Howard Braham',
+    'Kate Johnson',
+    'Brian Bergeron',
+  ],
+  'Extension UX': ['David Walsh', 'vthomas13', 'Nidhi Kumari', 'Victor Thomas'],
```
Comment (cortisiko) [inline scripts/generate-rc-commits.js:23]: Perhaps rename to Wallet UX?

File: scripts/generate-rc-commits.js [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+// JSON mapping authors to teams
+const authorTeams = {
+  Accounts: [
+    'Owen Craston',
+    'Gustavo Antunes',
+    'Monte Lai',
+    'Daniel Rocha',
+    'Howard Braham',
+    'Kate Johnson',
+    'Brian Bergeron',
```
Comment (cortisiko) [inline scripts/generate-rc-commits.js:21]: Add mike to this list as well. @plasmacorral

File: scripts/generate-rc-commits.js [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+    'yande',
+    'Aslau Mario-Daniel',
+  ],
+  WalletUX: [
+    'Frank von Hoven',
+  ],
+  Swaps: ['Daniel', 'Davide Brocchetto'],
+  Devex: ['Thomas Huang', 'Alex Donesky', 'jiexi', 'Zachary Belford'],
+  Security: ['witmicko'],
+  SDK: ['abretonc7s'],
```
Comment (cortisiko) [inline scripts/generate-rc-commits.js:105]: @christopherferreira9, @omridan159 @andreahaku as well for the SDK

File: scripts/generate-rc-commits.js [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+};
```
Comment (cortisiko) [inline scripts/generate-rc-commits.js:106]: can we also have a hardware wallet group as well? @vivek-consensys @stanleyyconsensys should be part of this group

File: scripts/generate-rc-commits.js [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+    'Sylva Elendu',
+    'seaona',
+  ],
+  'Confirmation Systems': [
+    'OGPoyraz',
+    'vinistevam',
+    'Matthew Walsh',
+    'cryptotavares',
+    'Vinicius Stevam',
+    'Derek Brans',
```
Comment (cortisiko) [inline scripts/generate-rc-commits.js:55]: add @sleepytanya to this list as well.

**Review by cortisiko — CHANGES_REQUESTED** [review]
left a few comments

File: scripts/generate-rc-commits.js [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+ */
+
+// JSON mapping authors to teams
+const authorTeams = {
+  Accounts: [
+    'Owen Craston',
+    'Gustavo Antunes',
+    'Monte Lai',
+    'Daniel Rocha',
+    'Howard Braham',
```
Comment (plasmacorral) [inline scripts/generate-rc-commits.js:19]: Howard is not a member of the accounts team

File: scripts/generate-rc-commits.js (hunk 3 above)
Comment (plasmacorral) [inline scripts/generate-rc-commits.js:21]: I think Brian is on the assets team

**Comment by jpcloureiro** [comment]
Is there anyone in charge of keeping the `JSON mapping authors to teams` up to date?

**Comment by sethkfman** [comment]
@jpcloureiro The delivery team will be updating this each time the run an analysis. It flags the unknown members and they can be added to the sheet.

File: scripts/generate-rc-commits.js [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+// JSON mapping authors to teams
+const authorTeams = {
+  Accounts: [
+    'Owen Craston',
+    'Gustavo Antunes',
+    'Monte Lai',
+    'Daniel Rocha',
+    'Kate Johnson',
+  ],
```
Comment (plasmacorral) [inline scripts/generate-rc-commits.js:21]: Ok, one more team member to add to accounts, please: Charly Chevalier

**Review by plasmacorral — DISMISSED** [review]
looks good to me, Accounts team members are accurate

**Comment by gantunesr** [comment]
Why certain names show up as their actual names and other as their GH handle? Which one is the correct?

**Comment by sethkfman** [comment]
@gantunesr It is based on the users git config and what is provided in the commit logs. Eventually we will be modifying the script to use team names.

**Comment by sethkfman** [comment]
@SocketSecurity ignore npm/simple-git@3.22.0

**Comment by sethkfman** [comment]
replaced by #9425

**Pull request closed without merging**